src/event.rs

```diff
@@ -225,107 +139,61 @@ pub enum WindowEvent<'a> {
     ///
     /// When the user drops multiple files at once, this event will be emitted for each file
     /// separately.
-    DroppedFile(PathBuf),
+    FileDropped(PathBuf),
 
     /// A file is being hovered over the window.
     ///
     /// When the user hovers multiple files at once, this event will be emitted for each file
     /// separately.
-    HoveredFile(PathBuf),
+    FileHovered(PathBuf),
 
     /// A file was hovered, but has exited the window.
     ///
     /// There will be a single `HoveredFileCancelled` event triggered even if multiple files were
     /// hovered.
-    HoveredFileCancelled,
+    FileHoverCancelled,
+
+    /// The window gained focus.
+    FocusGained,
+
+    /// The window lost focus.
+    FocusLost,
 
     /// The window received a unicode character.
-    ReceivedCharacter(char),
+    CharReceived(char),
 
-    /// The window gained or lost focus.
-    ///
-    /// The parameter is true if the window has gained focus, and false if it has lost focus.
-    Focused(bool),
-
-    /// An event from the keyboard has been received.
-    KeyboardInput {
-        device_id: DeviceId,
-        input: KeyboardInput,
-        /// If `true`, the event was generated synthetically by winit
-        /// in one of the following circumstances:
-        ///
-        /// * Synthetic key press events are generated for all keys pressed
-        ///   when a window gains focus. Likewise, synthetic key release events
-        ///   are generated for all keys pressed when a window goes out of focus.
-        ///   ***Currently, this is only functional on X11 and Windows***
-        ///
-        /// Otherwise, this value is always `false`.
-        is_synthetic: bool,
-    },
+    KeyPress(KeyPress),
```

Comment:
`KeyPress` is rather synonymous with "down", while `Release` is thought of as "up". Perhaps this should just be named `Key` ?

Reply:
A complicating factor here is that `KeyPress` and `PointerPress` should use the same verb to describe the keypress action, since the events represent similar actions. I've spent a lot of time off of github trying to figure out what this event should be called, and this was the best name I could come up with.

Another option would be to split those events into separate `Down`/`Up` events, though there are decent ergonomic reasons not to do that which @Ralith can do a better job of describing than I can.

Reply:
Maybe `Key` and `PointerButton`? That's pretty consistent with the web API terminology and less ambiguous.

A unified event is convenient for applications which need to dispatch both press and release events for a certain key along the same pathway. If they were separate events, entry into that pathway would need to be duplicated, which is less convenient and introduces opportunity for error.

Reply:
Those seem reasonable. I've renamed the events to those.